dev/sbt-plugin/src/main/scala/com/lightbend/lagom/sbt/InternalConfigs.scala

```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package com.lightbend.lagom.sbt
+
+import sbt._
+
+object InternalConfigs {
+  private[sbt] val devModeConfig = config("lagom-dev").hide
+  private[sbt] val cassandraDevModeConfig = config("lagom-dev-cassandra").hide
```

Comment:
Just to illustrate, if I were to upgrade netty to the latest 4.0 version to fix a bug, with the current implementation, there would be no guarantee that I'd get it in dev mode, since the `lagom-dev` configuration will also bring in netty, and then the `lagom-dev` classpath and the `runtime` classpath are simply concatenated to each other, and both versions of netty will be in that list.  Which one I get depends on implementation details of the Lagom plugin.

Whereas if it's done properly with ivy configuration dependencies, then ivy will see that in `runtime` I'm depending on a newer version of netty, and evict the older one that's brought in transitively through the `service-registry-client` `lagom-dev` dependency, so the classpath will once again only have one version of Netty, and it will be one decided predictably by Ivy.